common/url.go

```diff
@@ -110,10 +111,11 @@ type URL struct {
 	paramsLock sync.RWMutex
 	params     url.Values
 
-	Path     string // like  /com.ikurento.dubbo.UserProvider
-	Username string
-	Password string
-	Methods  []string
+	Path       string // like  /com.ikurento.dubbo.UserProvider
+	Username   string
+	Password   string
+	Methods    []string
+	MethodInfo []server.MethodInfo
```

Comment:
Maybe we should discuss with others about introducing new field into URL.

Reply:
Yes, we should revisit the necessity of adding this new field and its possible affect.

Reply:
I think we can use `URL.Attributes` to deliver `MethodInfo`, `URL.Attributes` is designed for delivering objects for runtime use.